modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/Relationships.tsx

```diff
@@ -240,9 +245,64 @@ export default function Relationships({
 		],
 	};
 
+	const onAfterAddObjectRelationship = async ({
+		objectDefinitionId1,
+	}: ObjectRelationship) => {
+		const toastMessage = Liferay.Language.get(
+			'relationship-was-created-successfully'
+		);
+
+		let toastAction;
+
+		if (objectDefinitionId !== objectDefinitionId1.toString()) {
+			toastAction = {
+				linkHref: await getEditObjectRelationshipURL(
+					baseResourceURL,
+					objectDefinitionId1
+				),
+				linkLabel: Liferay.Language.get('view-relationship'),
+			};
+		}
+
+		sessionStorage.setItem(
```

Comment:
Hey @cgmonte, can we think of other solutions instead of using localStorage?

What limitations can we observe when using a simple state for the toastAction?

Reply:
This a common pattern in Portal when you want to display a success toast to the user after a reload. Some examples:

https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/toasts.js#L32-L43

https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/components/DeletePublicationsFDSPropsTransformer.js#L21-L29

https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/js/edit_cp_definition_visibility.js#L137-L149

What is lacking here is setting as [CONSENT_TYPE_NECESSARY](/home/me/dev/projects/liferay-portal/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/consent.js), which I will add to this PR soon.

Reply:
Remember that the use of `sessionStorage` may present security flaws, including for storing a URL that will be accessed later by the user.

in the browser, a malicious user could change the destination URL, using something similar to this:

```
sessionStorage.setItem('addObjectRelationshipSuccessToast', ' {
    linkHref: 'www.anyurl.com',
    linkLabel: 'any content',
}');
```

I would just like to make my concerns regarding this clear.

Reply:
The browser restricts access to the same origin. From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API):

> sessionStorage maintains a separate storage area for each given [origin](https://developer.mozilla.org/en-US/docs/Glossary/Origin) that's available for the duration of the page session (as long as the browser tab is open, including page reloads and restores).

So, only a malicious script running on the same origin as Liferay, in the same tab (which basically means Liferay itself) could alter the data in a session storage. By the time a hacker hacked Liferay to execute malicious scripts and modified the session data, the door was already opened elsewhere, trough XSS or other means. In a situation like this, once there is malicious arbitrary code being executed, sessionStorage is only one among many opportunities to trick the user. Basically, in this scenario, any link in the page can be modified.